attester/src/az_snp_vtpm/mod.rs

```diff
@@ -7,11 +7,44 @@ use super::Attester;
 use anyhow::*;
 use az_snp_vtpm::{imds, vtpm};
 use log::debug;
+use raw_cpuid::{cpuid, CpuId, Hypervisor};
 use serde::{Deserialize, Serialize};
+use std::path::Path;
 
 pub fn detect_platform() -> bool {
-    if let Err(err) = vtpm::get_report() {
-        debug!("Failed to retrieve Azure HCL data from vTPM: {err}");
+    if !Path::new("/dev/tpm0").exists() {
+        debug!("vTPM device not found");
+        return false;
+    }
+
+    let cpuid = CpuId::new();
+    match cpuid.get_hypervisor_info() {
+        Some(hyper_info) => {
+            if hyper_info.identify() != Hypervisor::HyperV {
+                debug!("vTPM is only supported on Hyper-V");
+                return false;
+            }
+        }
+        None => {
+            debug!("Not a VM");
+            return false;
+        }
+    }
```

Comment:
Since we pass through the detection heuristics for each attester module (i.e. it "correctly" fails on other CSPs) this could add misleading output. Maybe we can make it less error-ish. (vTPMs also exist on GCP, afaik)

```suggestion
    let Some(hyper_info) = cpuid.get_hypervisor_info() else {
        debug!("Not a VM");
        return false;
    }   
    let hv_platform = hyper_info.identify();
    debug!("Hypervisor: {:?}", hv_platform;
    if hv_platform != Hypervisor::HyperV {
        return false;   
    }    
```